lib/db_utils.go

```diff
@@ -10661,6 +10680,483 @@ func DBDeletePostAssociationWithTxn(txn *badger.Txn, snap *Snapshot, association
+		LockupDurationNanoSecs: lockupDurationNanoSecs,
+	})
+
+	// Fetch the point from the database.
+	lockupYieldCurvePointBytes, err := DBGetWithTxn(txn, snap, key)
+	if err != nil {
+		return nil
+	}
+
+	// Parse the bytes beneath the key.
+	lockupYieldCurvePointObj := &LockupYieldCurvePoint{}
+	rr := bytes.NewReader(lockupYieldCurvePointBytes)
+	DecodeFromBytes(lockupYieldCurvePointObj, rr)
+	return lockupYieldCurvePointObj
+}
+
+func DBGetLocalYieldCurvePoints(handle *badger.DB, snap *Snapshot, profilePKID *PKID, lockupDurationNanoSecs int64) (
+	_leftLockupYieldCurvePoint *LockupYieldCurvePoint, _rightLockupYieldCurvePoint *LockupYieldCurvePoint) {
+	var leftLockupYieldCurvePoint *LockupYieldCurvePoint
+	handle.View(func(txn *badger.Txn) error {
+		leftLockupYieldCurvePoint = DBGetLeftLockupYieldCurvePointWithTxn(
+			txn, snap, profilePKID, lockupDurationNanoSecs)
+		return nil
+	})
+	var rightLockupYieldCurvePoint *LockupYieldCurvePoint
+	handle.View(func(txn *badger.Txn) error {
+		rightLockupYieldCurvePoint = DBGetRightLockupYieldCurvePointWithTxn(
+			txn, snap, profilePKID, lockupDurationNanoSecs)
+		return nil
+	})
+	return leftLockupYieldCurvePoint, rightLockupYieldCurvePoint
+}
+
+func DBGetLeftLockupYieldCurvePointWithTxn(txn *badger.Txn, snap *Snapshot, profilePKID *PKID,
+	lockupDurationNanoSecs int64) (_leftLockupYieldCurvePoint *LockupYieldCurvePoint) {
+	key := _dbKeyForLockupYieldCurvePoint(LockupYieldCurvePoint{
+		ProfilePKID:            profilePKID,
+		LockupDurationNanoSecs: lockupDurationNanoSecs,
+	})
+
+	// Seek left of the yield curve point.
+	iterLeftOpts := badger.DefaultIteratorOptions
+	iterLeftOpts.Reverse = true
+	iterLeft := txn.NewIterator(iterLeftOpts)
+	iterLeft.Seek(key)
+	iterLeftKey := iterLeft.Item().Key()
+
+	// There's a chance our seek yield a key in a different prefix (i.e. not a yield curve point).
+	// In this case, we know _dbKeyToLockupYieldCurvePoint will fail in parsing the key.
+	// We can return early in this case as there's no relevant yield points in the DB.
+	if len(iterLeftKey) < len(Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs) {
+		return nil
+	}
+	if !bytes.Equal(iterLeftKey[:len(Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs)],
+		Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs) {
+		return nil
+	}
+
+	// Fetch the LockupYieldCurvePoint beneath the key.
+	leftLockupYieldCurvePointBytes, err := DBGetWithTxn(txn, snap, iterLeftKey)
+	if err != nil {
+		return nil
+	}
+
+	// Parse the bytes beneath the key.
+	leftLockupYieldCurvePointObj := &LockupYieldCurvePoint{}
+	rr := bytes.NewReader(leftLockupYieldCurvePointBytes)
+	DecodeFromBytes(leftLockupYieldCurvePointObj, rr)
+	if !leftLockupYieldCurvePointObj.ProfilePKID.Eq(profilePKID) {
+		return nil
+	}
+
+	return leftLockupYieldCurvePointObj
+}
+
+func DBGetRightLockupYieldCurvePointWithTxn(txn *badger.Txn, snap *Snapshot, profilePKID *PKID,
```

Comment:
It's hard to be 100% sure all this is working so I'm just going to assume you tested it all extremely thoroughly.